share/completions/zabbix_server.fish

```diff
@@ -0,0 +1,105 @@
+set -l runtime config_cache_reload \
+    housekeeper_execute \
+    trigger_housekeeper_execute \
+    log_level_increase \
+    "log_level_increase=" \
+    log_level_decrease \
+    "log_level_decrease=" \
+    snmp_cache_reload \
+    secrets_reload \
+    diaginfo \
+    "diaginfo=" \
+    prof_enable \
+    prof_enable= \
+    prof_disable \
+    prof_disable= \
+    service_cache_reload \
+    ha_status \
+    "ha_remove_node=" \
+    ha_set_failover_delay 
+
+set -l scope rwlock mutex processing 
+
+
+function __fish_has_target 
+    if string match -rq '=' (commandline)
+        return 0
+    end
+    return 1
+end
+
+function __fish_string_in_command -a ch
+    if string match -rq $ch (commandline)
+        return 0
+    end
+    return 1
+end
+
+function __fish_prepend -a prefix
+    set -l log_target alerter \
+        "alert manager" \
+        "configuration syncer" \
+        discoverer \
+        escalator \
+        "history syncer" \
+        housekeeper \
+        "http poller" \
+        "icmp pinger" \
+        "ipmi manager" \
+        "ipmi poller" \
+        "java poller" \
+        poller \
+        "preprocessing manager" \
+        "preprocessing worker" \
+        "proxy poller" \
+        "self-monitoring" \
+        "snmp trapper" \
+        "task manager" \
+        timer \
+        trapper \
+        "unreachable poller" \
+        "vmware collector" \
+        "history poller" \
+        "availability manager" \
+        "service manager" \
+        "odbc poller"
+
+    if string match -rq 'log_level_(in|de)crease' $prefix
+        set var $log_target
+    else if string match -rq 'prof_(en|dis)able' $prefix
+        set var $log_target 'ha manager'
+    else if string match -rq 'diaginfo' $prefix
+        set var historycache preprocessing alerting lld valuecache locks
+    end
+
+    for i in $var
+        echo $prefix="$i"
+    end
+end
+
+
+function __fish_list_nodes
+    zabbix_server -R ha_status | tail -n+4 | awk '{print "ha_remove_node="$3}'
+end
+
+# General
+complete -c zabbix_server -s c -l config -d "Path to the configuration file"
+complete -c zabbix_server -f -s f -l foreground -d "Run Zabbix server in foreground"
+complete -c zabbix_server -f -s h -l help -d "Display this help message"
+complete -c zabbix_server -f -s V -l version -d "Display version number"
+
+
+# Log levels
+complete -c zabbix_server -r -f -s R -l runtime-control -n "__fish_has_target; __fish_string_in_command log_level_increase" -a "(__fish_prepend log_level_increase)"
```

Comment:
The status code of `__fish_has_target` will just be discarded here; you probably want to use a conjuction like &&.

Reply:
Figured that if it works without catching the exit cone, might as well ditch it.